src/Microsoft.AspNet.Mvc.Core/ApiExplorer/IApiResponseFormatMetadataProvider.cs

```diff
@@ -11,25 +11,18 @@ namespace Microsoft.AspNet.Mvc.ApiExplorer
     /// Provides metadata information about the response format to an <c>IApiDescriptionProvider</c>.
     /// </summary>
     /// <remarks>
-    /// An <see cref="Formatters.IOutputFormatter"/> should implement this interface to expose metadata information
+    /// An <see cref="IOutputFormatter"/> should implement this interface to expose metadata information
     /// to an <c>IApiDescriptionProvider</c>.
     /// </remarks>
     public interface IApiResponseFormatMetadataProvider
     {
         /// <summary>
-        /// Gets a filtered list of content types which are supported by the <see cref="Formatters.IOutputFormatter"/>
+        /// Gets a filtered list of content types which are supported by the <see cref="IOutputFormatter"/>
         /// for the <paramref name="declaredType"/> and <paramref name="contentType"/>.
         /// </summary>
-        /// <param name="declaredType">The declared type for which the supported content types are desired.</param>
-        /// <param name="runtimeType">The runtime type for which the supported content types are desired.</param>
-        /// <param name="contentType">
-        /// The content type for which the supported content types are desired, or <c>null</c> if any content
-        /// type can be used.
+        /// <param name="context">The <see cref="OutputFormatterCanWriteContext"/></param>
         /// </param>
-        /// <returns>Content types which are supported by the <see cref="Formatters.IOutputFormatter"/>.</returns>
-        IReadOnlyList<MediaTypeHeaderValue> GetSupportedContentTypes(
-            Type declaredType,
-            Type runtimeType,
-            MediaTypeHeaderValue contentType);
+        /// <returns>Content types which are supported by the <see cref="IOutputFormatter"/>.</returns>
+        IReadOnlyList<MediaTypeHeaderValue> GetSupportedContentTypes(OutputFormatterCanWriteContext context);
```

Comment:
It's now possible to reuse this context type here now, since we don't need the provide an `HttpContext`.